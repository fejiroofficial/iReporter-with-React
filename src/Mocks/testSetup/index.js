/* eslint-disable import/no-extraneous-dependencies */
import '@babel/polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';


export const mockState = {
  user: {
    token: '',
    isLoggedIn: false,
    loginStatus: false,
    message: '',
    isLoading: false,
    profile: {
      firstname: 'John',
      username: 'johnny',
      image: 'image.jpg',
      errorData: [],
    },
    resetPassword: {
      loading: false,
      message: '',
    }
  },
  bookmark: {
    loading: false,
    articles: [],
    message: '',
  },
  article: {
    newComments: {
      body: 'Hello world',
      id: 1,
      authorName: 'John',
      authorImage: 'image.jpg',
      date: 'July 4th 2018',
    },
    postingComment: false,
    commentMessage: 'Done',
    gettingComments: false,
    oldComments: [],
    hasMoreComments: false,
    commentPage: {
      current: 1,
      currentCount: 1,
      totalCount: 1,
    },
  },
  createArticle: {
    loading: false,
    message: false,
  },
  articlesReducer: {
    articles: []
  },
  articlesCategoryReducer: []
};


export const createMockStore = () => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  return mockStore;
};

const setup = (component, initailState = mockState) => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  const connectedWrapper = mount(
    <Provider store={mockStore(initailState)}>
      <BrowserRouter>
        {component}
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

export default setup;
