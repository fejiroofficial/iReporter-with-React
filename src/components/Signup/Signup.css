.SignUpContent {
    height: 780 px;
    width: 265px;
    margin-left: 26px;
    border-color: lightgrey;
    border-radius: 4px;
    border-style: solid;
    border-width: thin;
    margin-top: 50px;
}

input[class="input-field"] {
  width: inherit;
  padding: 12px 20px;
  font-size: 15px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  box-sizing: border-box;
  outline: none;
}

.SignupBox {
  width: inherit;
}

.SignUpHeader {
    height: 64px;
    width: inherit;
    text-align: center;
    font-size: x-large;
    padding-top: 12px;
    font-weight: bold;
}

.SignUpForm {
    margin: 0 auto;
    width: 229px;
    margin-bottom: 10px;
}

.SignBtn {
    height: 40px;
    width: 129px;
    border-width: thin;
    border-radius: 8px;
    margin-left: 54px;
    text-align: center;
    color: white;
    padding-top: 2px;
    background-color: lightseagreen;
    cursor: pointer;
    outline: none;
    border: none;
    font-weight: bold;
    font-size: 17px;
    margin-top: 5px;
}

.SignBtn:hover {
    background-color: #1f8983;
}

.SignNote {
    font-size: 14px;
    padding-left: 17px;
}

.SignNote a {
    margin-left: 4px;
    text-decoration: none;
    color: lightseagreen;
}

.server-info {
    background-color: dimgrey;
    min-height: 53px;
    position: absolute;
    display: none;
    width: -webkit-fill-available;
    color: white;
    text-align: -webkit-center;
    text-align: -moz-center;
}

#modal-info {
    margin-bottom: 22px;
}

.modal-log-area {
    display: flex;
    width: 310px;
    padding-top: 18px;
    padding-left: 12px;
    font-size: 11px;
}

.cancel-log {
    margin-left: 4px;
    height: 42px;
    width: 42px;
    margin-top: 0px;
    font-size: 12px;
    border-radius: 50%;
    font-weight: bold;
    cursor: pointer;
}

.spin-box {
    height: fit-content;
    width: inherit;
    text-align: -webkit-center;
    /** Mozilla browser window */
    text-align: -moz-center;
    padding-top: 16px;
}

#log-loader {
    display: none;
    position: absolute;
    margin-left: -27px;
    margin-top: 144px;
}

@media (min-width: 360px) {
    .SignUpContent {
        margin-left: 49px;
        margin-bottom: 83px;
    }
}

@media (min-width: 375px) {
    .SignUpContent {
        margin-left: 57px;
        margin-bottom: 107px;
    }
}

@media screen and (min-width: 375px) , screen and (min-height: 812px) {
    .SignUpContent {
        margin-left: 57px;
        margin-bottom: 277px;
    }
}

@media (min-width: 414px) {
    .SignUpContent {
        margin-left: 49px;
        width: 324px;
        margin-bottom: 178px;
    }
    .SignUpForm {
        width: 278px;
    }
    .SignBtn {
        margin-left: 76px;
    }
}

@media (min-width: 768px) {
    .modal-log-area {
        display: flex;
        width: 550px;
        padding-top: 11px;
        font-size: 20px;
    }
    .SignUpHeader {
        margin-bottom: 25px;
    }
    .SignUpContent {
        margin-left: 121px;
        width: 535px;
        margin-bottom: 307px;
    }
    .SignUpForm, #sign-up-form {
        width: 394px;
    }
    .SignBtn {
        margin-left: 136px;
    }
    .SignNote {
        padding-left: 70px;
    }
    #log-loader {
        display: none;
        position: absolute;
        margin-top: 124px;
    }
    .cancel-log {
        margin-top: -5px;
        font-size: 26px;
    }
}

@media (min-width: 1024px) {
    .SignUpContent {
        margin-left: 247px;
        margin-bottom: 482px;
    }
}


@media (min-width: 1440px) {
    .SignUpContent {
        margin-left: 467px;
        margin-bottom: 258px;
    }
    .display-images {
        height: 519px;
    }
}

@media (min-width: 1583px) {
    .SignUpContent {
        margin-left: 507px;
        margin-bottom: 258px;
    }
}