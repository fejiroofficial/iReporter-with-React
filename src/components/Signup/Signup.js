/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prefer-stateless-function */
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { successToast, failureToast } from '../../actions/toast';
import { signupUser } from '../../actions/auth';
import Classes from './Signup.css';
import { ToastContainer } from 'react-toastify';
import Spinner from '../spinner/Spinner';
// import 'react-toastify/dist/ReactToastify.css';

const signUp = (props) => {

  const [userData, setUserData] = useState({
    username: null,
    firstname: null,
    lastname: null,
    othernames: null,
    email: null,
    telephone: null,
    password: null,
  });
  const [loading, setLoading] = useState(false);

  const updateInput = event => {
    setUserData({ ...userData, [event.target.name]: event.target.value });
  };

  const completeSignup = () => {
    props.signupUser(userData).then(res => {
      const { errors } = res;
      if (errors) {
        setLoading(false);
        Object.values(errors).forEach(error => props.failureToast(error));
      }
      else if (res.success === 'false') {
        setLoading(false);
        props.failureToast(res.message);
      }
      else {
        props.history.push('/');
      }
    });
  };

  const validateUserData = (event) => {
    event.preventDefault();
    setLoading(true);
    const fieldsArr = Object.entries(userData);
    const errors = [];
    fieldsArr.forEach(field => {
      if (!field[1]) {
        errors.push(`${field[0]} is required.`);
        props.failureToast(`${field[0]} is required.`);
      }
    });
    return errors.length < 1 ? completeSignup() : setLoading(false);
  };

  return (
    <div className={Classes.SignupBox}>
      <div className={Classes.SignUpContent}>
        <ToastContainer autoClose={5000} />
        <div className={Classes.SignUpHeader}>Create an Account</div>
        <form className={Classes.SignUpForm}>
          <label for='Username'>Username</label>
          <input type='text' name='username' className="input-field" placeholder="Enter username" required onChange={(event) => updateInput(event)}></input>
          <label for='firstname'>First name</label>
          <input type='text' name='firstname' className="input-field" placeholder="Enter first name" required onChange={(event) => updateInput(event)}></input>
          <label for='lastname'>Last name</label>
          <input type='text' name='lastname' className="input-field" placeholder="Enter last name" required onChange={(event) => updateInput(event)}></input>
          <label for='othernames'>Other names</label>
          <input type='text' name='othernames' className="input-field" placeholder="Enter other names" required onChange={(event) => updateInput(event)}></input>
          <label for='email'>Email</label>
          <input type='text' name='email' className="input-field" placeholder="Enter email" required onChange={(event) => updateInput(event)}></input>
          <label for='telephone'>Telephone</label>
          <input type='text' name='telephone' className="input-field" placeholder="Enter telephone" required onChange={(event) => updateInput(event)}></input>
          <label for='Password'>Password</label>
          <input type='text' name='password' className="input-field" id='log-password' placeholder="Enter Password" required onChange={(event) => updateInput(event)}></input>
          <button
            className={Classes.SignBtn}
            onClick={(event) => validateUserData(event)}
            disabled={loading}
          >
            {loading ? <Spinner loading={loading} /> : 'Sign up'}
          </button>
        </form>
        <div className={Classes.SignNote}>
          <p>Do not have an account? Create an account
          <Link to='/login'>here...</Link>
          </p>
        </div>
      </div>
    </div>
  );
};

export default connect(() => ({}), { successToast, failureToast, signupUser })(signUp);
