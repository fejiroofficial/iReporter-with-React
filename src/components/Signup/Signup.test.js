import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { createMockStore } from '../../Mocks/testSetup';
import Signup from './Signup';

createMockStore();

describe('<Signup>', () => {
  it('should render without crashing', () => {
    const container = (
      <Signup>
        <p>Hello</p>
      </Signup>
    );
    const wrapper = shallow(container);
    expect(wrapper).toMatchSnapshot();
  });
});
