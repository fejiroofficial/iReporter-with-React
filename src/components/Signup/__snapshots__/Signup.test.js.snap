// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup> should render without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
